src/createcompendia/leftover_umls.py

```diff
@@ -0,0 +1,239 @@
+
+        with open('babel_outputs/reports/umls-types.tsv', 'w') as outf:
+            for tui in sorted(types_by_tui.keys()):
+                for sty in sorted(list(types_by_tui[tui])):
+                    outf.write(f"{tui}\t{sty}\n")
+
+        # Create a compendium that consists solely of all MRCONSO entries that haven't been referenced.
+        # Code adapted from datahandlers.umls.build_sets()
+        count_no_umls_type = 0
+        count_multiple_umls_type = 0
+        with open(mrconso, 'r') as inf:
+            for line in inf:
+                x = line.strip().split('|')
+                cui = x[0]
+                umls_id = f"{UMLS}:{cui}"
+                if umls_id in umls_ids_in_other_compendia:
+                    logging.debug(f"UMLS ID {umls_id} is in another compendium, skipping.")
+                    continue
+                if umls_id in umls_ids_in_this_compendium:
+                    logging.debug(f"UMLS ID {umls_id} has already been included in this compendium, skipping.")
+                    continue
+                lang = x[1]
+
+                # TODO: note that this technically skips terms that don't have any English labels.
+                #Only keep english terms
+                if lang != 'ENG':
+                    continue
+
+                #only keep unsuppressed rows
+                suppress = x[16]
+                if suppress == 'O' or suppress == 'E':
+                    continue
+                #only keep sources we're looking for
+                source = x[11]
+                # if source not in lookfor:
+                #     continue
+                # tty = x[12]
+
+                # The STR value should be the label.
+                str = x[14]
+
+                # Lookup type.
+                def umls_type_to_biolink_type(umls_tui):
+                    biolink_type = biolink_toolkit.get_element_by_mapping(f'STY:{umls_tui}', most_specific=True, formatted=True, mixin=True)
+                    if biolink_type is None:
+                        logging.debug(f"No Biolink type found for UMLS TUI {umls_tui}")
+                    return biolink_type
+
+                umls_type_results = types_by_id.get(umls_id, {'biolink:NamedThing': {'Named thing'}})
+                biolink_types = set(list(map(umls_type_to_biolink_type, umls_type_results.keys())))
+
+                # How to deal with multiple Biolink types? We currently only have the following multiple
+                # types, so we can resolve these manually:
+                biolink_types_as_str = '|'.join(sorted(map(lambda t: "(None)" if t is None else t, list(biolink_types))))
+                if None in biolink_types:
+                    # One of the TUIs couldn't be converted; let's delete all of them so that we can report this.
+                    biolink_types = list()
+
+                # Some Biolink multiple types we handle manually.
+                if biolink_types_as_str == 'biolink:Device|biolink:Drug':
```

Comment:
I think it would be good to use the constants in src/categories.py.   The goal is to insulate against name changes, though to be fair I don't know that's a big problem w/biolink.

Reply:
Fixed in 6db455aee8b2adce.